preparer.py

```diff
@@ -42,18 +51,21 @@ def prepare(self, client, collection_path_guest, generate_save_tag):
                 "mode": "ro"
             }
 
+        index_args = {
+            "collections": [{"name": name, "path": name_to_path_guest[name], "format": format} for (name, path, format) in collections]
+        }
+
         # The first step is to pull an image from an OSIRRC participant,
         # start up a container, run its `init` and `index` hooks, and then
         # use `docker commit` to save the image after the index has been
         # built. The rationale for doing this is that indexing may take a
         # while, but only needs to be done once, so in essence we are
         # "snapshotting" the system with the indexes.
-        base = client.containers.run("{}:{}".format(self.config.repo, self.config.tag),
-                                     command="sh -c '/init; /index --collections {}'".format(" ".join(name_to_path_host.keys())),
-                                     volumes=volumes, detach=True)
+        container = client.containers.run("{}:{}".format(self.config.repo, self.config.tag),
+                                     command="sh -c '/init; /index --json {}'".format(json.dumps(json.dumps(index_args))), volumes=volumes, detach=True)
```

Comment:
Double `dumps` is on purpose to ensure proper escaping when being passed to the `index` and `search` scripts.